pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -686,6 +695,44 @@ func checkForPodSecondaryNetworkAttachment(pod *corev1.Pod) (string, bool) {
 	return netObj, netObjExists && netObj != ""
 }
 
+// When a Kubernetes Node reboots and the OVSDB file is not properly restored,
+// the primary and secondary OVS ports in OVSDB are lost. Pods may temporarily
+// appear as "unknown" and later get recreated on the same Node with new container IDs.
+// The cleanup of secondary interfaces during agent restart depends on the InterfaceStores
+// being initialized with existing OVS ports. If the OVSDB file is missing, the
+// primary InterfaceStore is empty, preventing the secondary InterfaceStore from
+// initializing correctly. Consequently, secondary IPs assigned to old Pods (with
+// previous container IDs) are not deleted. To resolve this, all IPPool allocations
+// must be checked, and any IPs associated with non-existing container IDs should be
+// released. Otherwise, stale secondary IPs remain in the IPPool and cannot be reused.
+// We run it periodically to cover the case where a Pod is recreated with
+// the same name on another Node and has any secondary IP as well.
```

Comment:
I feel like this could be handled by using the Pod UID as the owner (instead of name), but that's probably a more impactful change